test/posh/lib/datascript_test.cljc

```diff
@@ -141,3 +141,138 @@
+        [:name :parent/score {:parent/child [:name {:parent/child [:name]}]}]
+        [:parent/id "grandma"]
+        '{:datoms {:posh ([1 :name "margit"]
+                          [1 :parent/score 16]
+                          [1 :parent/child 2]
+                          [2 :name "make"]
+                          [2 :parent/child 3]
+                          [3 :name "eeva"])},
+          :patterns {:posh ([#{1 2} :parent/child _]
+                            [#{3 2} #{:name} _]
+                            [#{1} #{:name :parent/score} _]
+                            [_ :parent/id "grandma"])}}
+
+        ;; _child -> _child
+        [:name {:parent/_child [:name {:parent/_child [:name :parent/score]}]}]
+        [:child/id "kid"]
+        '{:datoms {:posh ([3 :name "eeva"]
+                          [2 :parent/child 3]
+                          [2 :name "make"]
+                          [1 :parent/child 2]
+                          [1 :name "margit"]
+                          [1 :parent/score 16])},
+          :patterns {:posh ([_ :parent/child 2]
+                            [_ :parent/child 3]
+                            [#{1} #{:name :parent/score} _]
+                            [#{3 2} #{:name} _]
+                            [_ :child/id "kid"])}}))))
+
+(deftest pull-reverse-many-test
+  (let [schema {:parent/id {:db/unique :db.unique/identity}
+                :parent/child {:db/valueType :db.type/ref
+                               :db/isComponent false
+                               :db/cardinality :db.cardinality/many}
+                :child/id {:db/unique :db.unique/identity}}
+        data [{:parent/id "grandma"
+               :parent/score 16
+               :name "margit"
+               :parent/child [{:parent/id "mom"
+                               :name "irmeli"
+                               :parent/child [{:child/id "kid"
+                                               :name "eeva"}]}
+                              {:parent/id "dad"
+                               :name "make"
+                               :parent/child [{:child/id "kid"
+                                               :name "eeva"}]}]}]]
+
+    (testing "analysis"
+      (are [pattern id expected]
+        (let [ana (analyze schema data pattern id)]
+          (is (= expected ana)))
+
+        ;; child -> child -> _child -> _child
+        [:name {:parent/child [:name {:parent/child [:name {:parent/_child [{:parent/_child [:parent/score]}]}]}]}]
+        [:parent/id "grandma"]
+        '{:datoms {:posh ([1 :name "margit"]
+                          [1 :parent/child 2]
+                          [1 :parent/child 4]
+                          [2 :name "irmeli"]
+                          [2 :parent/child 3]
+                          [3 :name "eeva"]
+                          [1 :parent/score 16]
+                          [4 :parent/child 3]
+                          [4 :name "make"])},
+          :patterns {:posh ([_ :parent/child 3]
```

Comment:
note: these could be accumulated into `[_ :parent/child #{2 3 4}]` in the future.